Cuts from cell type annotation results section (#179)
AlexsLemonade/ScPCA-manuscript: content/03.results.md

Assigning cell type labels to single-cell and single-nuclei RNA-seq data is often an essential step in analysis.
Cell type annotation requires knowledge of the expected cell types in a dataset and associated gene expression patterns for each cell type, which may be available in other public databases or individual publications.
Automated cell type annotation methods leveraging public databases are an excellent initial step in the labeling process, as they can be applied consistently and transparently across all samples in a dataset.
As such, we include cell type annotations determined using two different automated methods, `SingleR` [@doi:10.1038/s41590-018-0276-y] and `CellAssign` [@doi:10.1038/s41592-019-0529-1], in all processed `SingleCellExperiment` and `AnnData` objects.

> Most public annotated reference datasets that can be used with `SingleR` and `CellAssign`, including those we use for the Portal, are derived from normal tissue, making accurately annotating tumor datasets particularly difficult.

Author: Okay fair enough
```suggestion
Most public annotated reference datasets that can be used with `SingleR` and `CellAssign` – including those we use for the Portal – are derived from normal tissue, making accurately annotating tumor datasets particularly difficult.
```

We calculate the Jaccard index between pairs of cell type labels to compare annotations between the two methods and display it in a heatmap (an example is shown in Figure {@fig:fig4}B).
Jaccard index values close to 1 indicate high agreement and a high proportion of overlapping cells, which may indicate higher confidence predictions.

The report also includes diagnostic plots for each method.
To evaluate confidence in `SingleR` cell type annotations, the delta median statistic is calculated by subtracting the median score from the score associated with the assigned cell type label [@url:https://bioconductor.org/books/release/SingleRBook/annotation-diagnostics.html#based-on-the-deltas-across-cells].
The cell type report shows the distribution of delta median values for each cell type. 
A higher delta median statistic for a cell indicates higher confidence in the final cell type annotation.
We also display the distribution of all probabilities calculated by `CellAssign`; more confident labels are expected to have many values close to 1.

> If the submitter provided cell types, the submitter annotations are compared to the annotations from both `SingleR` and `CellAssign`.

Reviewer: Replaced by the suggestion below
```suggestion
```

> The cell type report includes a summary of this comparison, a table summarizing the submitter cell type annotations, and a UMAP plot in which each cell is colored by the submitter annotation.

Reviewer: The order of descriptions here seemed odd. So I tried combining all parts. It's probably too long now.
```suggestion
If the submitter provided cell type annotations, the cell type report also includes a table summarizing the submitter cell type annotations, a UMAP plot in which each cell is colored by the submitter annotation, and a comparison of the submitter annotations to the automated cell typing results from `SingleR` and `CellAssign`.
```

The Jaccard index is calculated for all pairs of cell type labels in submitter annotations and `SingleR` annotations, and in submitter annotations and `CellAssign` annotations.
The results from both comparisons are displayed in a stacked heatmap available in the report, an example of which is shown in Figure {@fig:figS5}B.

### Assigning consensus cell types

`SingleR` and `CellAssign` use different references and distinct computational approaches to label cells.
We expect cells with the same or similar cell type labels from both methods to be more accurately annotated.
`scpca-nf` therefore assigns consensus cell type labels when the two automated methods agree.
To account for different levels of granularity in reference datasets, we employed an ontology-based approach to assign a consensus cell type label.
Specifically, the consensus cell type annotation is equivalent to the latest common ancestor (LCA) in Cell Ontology [@doi:10.1186/s13326-016-0088-7; @doi:10.1186/1471-2105-12-6; @doi: 10.1186/gb-2005-6-2-r21] shared between the two predicted cell types.

> To ensure specificity in the consensus labels, cells were only assigned a consensus cell type if the identified LCA had no more than 170 descendant terms, with a few exceptions (see Materials and Methods for more details).

Reviewer: Noting here that in earlier edits we refer only to methods
```suggestion
To ensure specificity in the consensus labels, cells were only assigned a consensus cell type if the identified LCA had no more than 170 descendant terms, with a few exceptions (see Methods for more details).
```
